File: Homework/05-Global-Search/python/queens_bfs_dfs.py
from time import time

from algorithm.breadth_first_search import BreadthFirstSearch
from algorithm.depth_first_search import DepthFirstSearch
from problem.queens import QueensState

if __name__ == "__main__":
    # t0 = time()

    # s = QueensState(12)

    # bfs = BreadthFirstSearch(s)

    # bfs.search(True, False)

    # dfs = DepthFirstSearch(s)

    # dfs.search(True, False)
    for i in range(8, 14):
        t0 = time()
        print(f"Queens State for {i} queens:")
        s = QueensState(i)
        bfs = BreadthFirstSearch(s)
        bfs.search(True, False)
        print(f"\tBFS time = {time() - t0}s")
        t0 = time()
        dfs = DepthFirstSearch(s)
        dfs.search(True, False)
        print(f"\tDFS time = {time() - t0}s")
        print("----")

    # print(f"time = {time() - t0}s")

File: Homework/05-Global-Search/c++/queens_bfs_dfs.cpp
#include <iostream>
#include <ctime>

#include "problem/queens.hpp"

#include "algorithm/depth_first_search.hpp"
#include "algorithm/breadth_first_search.hpp"

int main(){
    std::ios::sync_with_stdio(false);

    // time_t t0 = time(nullptr);
    
    // QueensState state(12);
    // BreadthFirstSearch<QueensState> bfs(state);
    // bfs.search(true, false);

    // //DepthFirstSearch<QueensState> dfs(state);
    // //dfs.search(true, false);
    
    // std::cout << time(nullptr) - t0 << std::endl;
    // return 0;

    for (int i = 8; i <= 15; ++i) {
        std::cout << "Queens State for " << i << " queens:" << std::endl;

        time_t t0 = time(nullptr);
        QueensState state(i);

        BreadthFirstSearch<QueensState> bfs(state);
        bfs.search(true, false);
        std::cout << "\tBFS time = " << time(nullptr) - t0 << "s" << std::endl;

        t0 = time(nullptr);
        DepthFirstSearch<QueensState> dfs(state);
        dfs.search(true, false);
        std::cout << "\tDFS time = " << time(nullptr) - t0 << "s" << std::endl;

        std::cout << "----" << std::endl;
    }
}
